### PR TITLE
Allow for 0 existential deposit in benchmarks for `pallet_staking`, `pallet_session`, and `pallet_balances`

### PR DESCRIPTION
**Pull request by krisbitney**
This PR ensures non-zero values are available in benchmarks for `pallet_staking`, `pallet_session`, and `pallet_balances` where required for them to run.

This small change makes it possible to run the benchmarks for `pallet_staking`, `pallet_session`, and `pallet_balances` in a runtime for which existential deposit is set to 0. 

The benchmarks for `pallet_staking` and `pallet_session` will still fail in runtimes that use `U128CurrencyToVote`, but that is easy to work around by creating a new `CurrencyToVote` implementation for benchmarking.

The changes are implemented by checking if existential deposit equals 0 and using 1 if so.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot fmt

**Comment by bkchr** [comment]
@krisbitney could you please write a [prdoc](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md)

**Review by ggwpez — APPROVED** [review]

**Review by gpestana — APPROVED** [review]

**Comment by krisbitney** [comment]
> @krisbitney could you please write a [prdoc](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md)

Done!

File: prdoc/pr_4346.prdoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+doc:
+  - audience: Runtime Dev
+    description: |
+      Changes were made to benchmarks for `pallet_staking`, `pallet_session`, and `pallet-balances` to accommodate runtimes with 0 existential deposit. This should not affect the vast majority of runtimes. For runtimes with 0 existential deposit, the benchmarks for `pallet_staking` and `pallet_session` will still fail when using `U128CurrencyToVote` in the `pallet-staking` config; developers can use or write another `CurrencyToVote` implementation for benchmarking to work around this.
+
+crates:
+  - name: pallet-staking
+    bump: patch
+  - name: pallet-session
+    bump: patch
```
Comment (bkontur) [inline prdoc/pr_4346.prdoc:15]: maybe I am missing something, but I don't see any changes to pallet-session here?

Comment (krisbitney) [reply prdoc/pr_4346.prdoc]: I just pushed a commit change it to `pallet-session-benchmarking`

**Review by bkchr — APPROVED** [review]

**Review by bkontur — APPROVED** [review]

**Pull request merged**